packages/splits-sdk/src/client/splitV2.ts

```diff
@@ -0,0 +1,820 @@
+  }
+
+  protected async _setPaused({
+    split,
+    paused,
+    transactionOverrides = {},
+  }: SetPausedConfig): Promise<TransactionFormat> {
+    validateAddress(split)
+
+    this._requirePublicClient()
+    if (this._shouldRequireWalletClient) this._requireWalletClient()
+
+    return this._executeContractFunction({
+      contractAddress: split,
+      contractAbi: splitV2ABI,
+      functionName: 'setPaused',
+      functionArgs: [paused],
+      transactionOverrides,
+    })
+  }
+
+  protected async _execCalls({
+    splitAddress,
+    calls,
+    transactionOverrides = {},
+  }: SplitV2ExecCallsConfig): Promise<TransactionFormat> {
+    validateAddress(splitAddress)
+    calls.map((call) => validateAddress(call.to))
+
+    this._requirePublicClient()
+    if (this._shouldRequireWalletClient) this._requireWalletClient()
+
+    return this._executeContractFunction({
+      contractAddress: splitAddress,
+      contractAbi: splitV2ABI,
+      functionName: 'execCalls',
+      functionArgs: [calls],
+      transactionOverrides,
+    })
+  }
+
+  protected async _updateSplit({
+    split,
+    recipients,
+    allocations,
+    distributionIncentive,
+    transactionOverrides = {},
+  }: UpdateSplitV2Config): Promise<TransactionFormat> {
+    validateAddress(split)
+    recipients.map((recipient) => validateAddress(recipient))
+
+    this._requirePublicClient()
+    if (this._shouldRequireWalletClient) this._requireWalletClient()
+
+    const totalAllocation = allocations.reduce((a, b) => a + b)
+
+    return this._executeContractFunction({
+      contractAddress: split,
+      contractAbi: splitV2FactoryABI,
+      functionName: 'updateSplit',
+      functionArgs: [
+        {
+          recipients,
+          allocations,
+          totalAllocation,
+          distributionIncentive,
+        },
+      ],
+      transactionOverrides,
+    })
+  }
+
+  protected async _distribute({
+    split,
+    recipients,
+    allocations,
+    distributionIncentive,
+    token,
+    distributor,
+    transactionOverrides = {},
+  }: DistributeSplitConfig): Promise<TransactionFormat> {
+    validateAddress(split)
+    validateAddress(token)
+    validateAddress(distributor)
+    recipients.map((recipient) => validateAddress(recipient))
+
+    this._requirePublicClient()
+    if (this._shouldRequireWalletClient) this._requireWalletClient()
+
+    const totalAllocation = allocations.reduce((a, b) => a + b)
+
+    return this._executeContractFunction({
+      contractAddress: split,
+      contractAbi: splitV2FactoryABI,
+      functionName: 'distribute',
+      functionArgs: [
+        {
+          recipients,
+          allocations,
+          totalAllocation,
+          distributionIncentive,
+        },
+        token,
+        distributor,
+      ],
+      transactionOverrides,
+    })
+  }
+
+  protected _getSplitV2Contract(
+    split: Address,
+  ): GetContractReturnType<SplitV2ABI, PublicClient<Transport, Chain>> {
+    return getContract({
```

Comment:
nit: you should rebase this off the `v4` branch. It's using viem v2 and `getContract` has changed slightly (it's basically the same though)